src/components/List/List.tsx

```diff
@@ -171,58 +83,57 @@ export class List extends React.Component {
         }
     }
 
-    refFilter = React.createRef();
-    refContainer = React.createRef();
-    getContainer() {
-        const ref = this.refContainer.current;
-        return this.props.sortable ? ref && ref.getWrappedInstance() : ref;
-    }
+    render() {
+        const {emptyPlaceholder, virtualized, className, itemsClassName} = this.props;
 
-    getActiveItem() {
-        return typeof this.state.activeItem === 'number' ? this.state.activeItem : null;
+        const {items} = this.state;
+
+        return (
+            <MobileContext.Consumer>
+                {({mobile}) => (
+                    <div
+                        className={b({mobile}, className)}
+                        tabIndex={-1}
+                        onFocus={this.onFocus}
+                        onBlur={this.onBlur}
+                        onKeyDown={this.onKeyDown}
+                    >
+                        {this.renderFilter()}
+                        <div
+                            className={b('items', {virtualized}, itemsClassName)}
+                            style={this.getItemsStyle()}
+                            onMouseLeave={this.onMouseLeave}
+                        >
+                            {this.renderItems()}
+                            {items.length === 0 && Boolean(emptyPlaceholder) && (
+                                <div className={b('empty-placeholder')}>{emptyPlaceholder}</div>
+                            )}
+                        </div>
+                    </div>
+                )}
+            </MobileContext.Consumer>
+        );
     }
 
-    filterItem = (filter) => (item) => {
-        return String(item).includes(filter);
-    };
     getItems() {
         return this.state.items;
     }
-    getFilter() {
-        const {filter = this.state.filter} = this.props;
-        return filter;
-    }
-    getItemsStyle() {
-        let {itemsHeight} = this.props;
-        if (typeof itemsHeight === 'function') {
-            itemsHeight = itemsHeight(this.state.items);
-        }
-        return itemsHeight ? {height: itemsHeight} : undefined;
+
+    getActiveItem() {
+        return typeof this.state.activeItem === 'number' ? this.state.activeItem : null;
     }
-    scrollToIndex = (index) => {
-        const container = this.getContainer();
-        if (container) {
-            container.scrollToItem(index);
-        }
-    };
-    activateItem(index, scrollTo = true) {
-        if (scrollTo) {
+
+    activateItem(index?: number, scrollTo = true) {
+        if (index && scrollTo) {
```

Comment:
I added `index` to condition because of uselessness `scrollToIndex` without it.

Reply:
0 will be treated as false, better change to `typeof index === 'number'`